src/flow-manager.c

```diff
@@ -248,6 +249,26 @@ static int FlowManagerFlowTimeout(const Flow *f, enum FlowState state, struct ti
 
     /* do the timeout check */
     if (flow_times_out_at >= ts->tv_sec) {
+        /* for bypassed flow we need to check if the capture method has seen
+         * something and update the logic */
+        FlowBypassInfo *fc = FlowGetStorageById(f, GetFlowBypassInfoID());
```

Comment:
should we wrap this in if (flow state is non-local bypass) { ?

Reply:
Yes, good catch.

Reply:
In fact, we will have the storage if and only if we have a capture bypassed flow so IMO we should keep it like that.

Reply:
Right ok. Although checking 'state' (a local variable) is probably cheaper than a memory access at the end of the structure.